File: examples/run_catalog_actor.py
"""Run any public Arclay Actor and print a sample of its dataset rows.

Usage:
    python examples/run_catalog_actor.py \
        arclay-tn/eu-funding-calls-intelligence \
        '{"searchQuery":"digital","maxItems":10}'

Set APIFY_TOKEN in the environment before running.
"""

from __future__ import annotations

import argparse
import json
import os

from apify_client import ApifyClient


def main() -> None:
    parser = argparse.ArgumentParser()
    parser.add_argument("actor_id", help="Apify username/actor-name")
    parser.add_argument("input_json", help="Actor input as a JSON object")
    args = parser.parse_args()

    run_input = json.loads(args.input_json)
    if not isinstance(run_input, dict):
        raise ValueError("input_json must decode to a JSON object")

    client = ApifyClient(os.environ["APIFY_TOKEN"])
    run = client.actor(args.actor_id).call(run_input=run_input)
    items = list(client.dataset(run["defaultDatasetId"]).iterate_items())
    print(json.dumps(items[:10], indent=2, ensure_ascii=False))


if __name__ == "__main__":
    main()

File: examples/run_catalog_actor.mjs
// Run any public Arclay Actor and print a sample of its dataset rows.
// npm install
// APIFY_TOKEN=... node examples/run_catalog_actor.mjs \
//   arclay-tn/eu-funding-calls-intelligence '{"searchQuery":"digital","maxItems":10}'

import { ApifyClient } from "apify-client";

const [actorId, inputJson] = process.argv.slice(2);
if (!actorId || !inputJson) {
  throw new Error("Usage: node examples/run_catalog_actor.mjs ACTOR_ID INPUT_JSON");
}

const input = JSON.parse(inputJson);
if (!input || typeof input !== "object" || Array.isArray(input)) {
  throw new Error("INPUT_JSON must decode to a JSON object");
}

const client = new ApifyClient({ token: process.env.APIFY_TOKEN });
const run = await client.actor(actorId).call(input);
const { items } = await client.dataset(run.defaultDatasetId).listItems();
console.log(JSON.stringify(items.slice(0, 10), null, 2));
